tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/DeepEqualsTest.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.test.nativeimpl.functions;
+
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.codegen.FunctionInfo;
+import org.ballerinalang.util.codegen.PackageInfo;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.FileNotFoundException;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * Test cases for ballerina.reflect:deepEquals() function.
+ */
+public class DeepEqualsTest {
+    CompileResult compileResultForPrimitives;
+    CompileResult compileResultForComplex;
```

Comment:
Shall we make above two private?

Reply:
Fixed with https://github.com/ballerina-lang/ballerina/pull/4680/commits/4565dd5c6f715a95bf789168751ed0217a6d5631